src/request/sign-swap/SignSwapApi.js

```diff
@@ -0,0 +1,132 @@
+/* global BitcoinRequestParserMixin */
+/* global TopLevelApi */
+/* global Nimiq */
+/* global SignSwap */
+/* global Errors */
+
+class SignSwapApi extends BitcoinRequestParserMixin(TopLevelApi) {
+    /**
+     * @param {KeyguardRequest.SignSwapRequest} request
+     * @returns {Promise<Parsed<KeyguardRequest.SignSwapRequest>>}
+     */
+    async parseRequest(request) {
+        if (!request) {
+            throw new Errors.InvalidRequestError('request is required');
+        }
+
+        try {
+            sessionStorage.setItem('_test', 'write-access');
+            const stored = sessionStorage.getItem('_test');
+            if (stored !== 'write-access') throw new Error();
+            sessionStorage.removeItem('_test');
+        } catch (e) {
+            throw new Error('Cannot access browser storage because of privacy settings');
+        }
+
+        /** @type {Parsed<KeyguardRequest.SignSwapRequest>} */
+        const parsedRequest = {};
+        parsedRequest.appName = this.parseAppName(request.appName);
+        parsedRequest.keyInfo = await this.parseKeyId(request.keyId);
+        if (parsedRequest.keyInfo.type !== Nimiq.Secret.Type.ENTROPY) {
+            throw new Errors.InvalidRequestError('Bitcoin is only supported with modern accounts.');
+        }
+        parsedRequest.keyLabel = this.parseLabel(request.keyLabel, true, 'keyLabel');
+
+        parsedRequest.swapId = /** @type {string} */ (this.parseLabel(request.swapId, false, 'swapId'));
+
+        if (request.fund.type === request.redeem.type) {
+            throw new Errors.InvalidRequestError('Swap must be between two different currencies');
+        }
+
+        if (request.fund.type === 'NIM') {
+            parsedRequest.fund = {
+                type: 'NIM',
+                keyPath: this.parsePath(request.fund.keyPath, 'fund.keyPath'),
+                transaction: this.parseTransaction({
+                    validityStartHeight: 0, // Dummy
+                    data: new Uint8Array(78), // Dummy
+                    ...request.fund,
+                    flags: Nimiq.Transaction.Flag.CONTRACT_CREATION,
+                    recipient: 'CONTRACT_CREATION',
+                    recipientType: Nimiq.Account.Type.HTLC,
```

Comment:
Here we set the fixed properties that we _know_ must be when an HTLC is supposed to be created.